tests/lib/check-component-out-of-board.test.ts

```diff
@@ -0,0 +1,21 @@
+import { test, expect } from "bun:test"
+import componentOutOfBoard from "tests/assets/component-out-of-board.json"
+import { checkPcbComponentsOutOfBoard } from "lib/check-pcb-components-outside-board/areComponentsOutOfBoard"
+test("pcb component out of board", async () => {
+  const pcbPlacementError = checkPcbComponentsOutOfBoard(
+    componentOutOfBoard as any,
+  )
+
+  expect(pcbPlacementError).toMatchInlineSnapshot(
+    `
+  [
+    {
+      "message": "Component source_component_1 out of board",
```

Comment:
you should use a proper display name for the component, ids are very unfriendly for users

Reply:
a proper display name could be e.g. `resistor["R1"]`